x-pack/plugins/transform/server/routes/api/error_utils.ts

```diff
@@ -109,14 +109,14 @@ function extractCausedByChain(
  * @return Object Boom error response
  */
 export function wrapEsError(err: any, statusCodeToMessageMap: Record<string, any> = {}) {
-  const { statusCode, response } = err;
+  const { statusCode, body } = err;
```

Comment:
Tried creating a transform with an invalid aggregation by using the JSON editor, and some details of the error are getting lost compared to the previous behavior.

Before (preview):
<img width="881" alt="transforms_error3" src="https://user-images.githubusercontent.com/7405507/101512904-77458880-3973-11eb-81f0-57df18a21bd6.PNG">


Before (create):
<img width="413" alt="transforms_error4" src="https://user-images.githubusercontent.com/7405507/101512930-7ca2d300-3973-11eb-8999-af217ab1611d.PNG">


After (preview):
<img width="866" alt="transforms_error1" src="https://user-images.githubusercontent.com/7405507/101512969-8593a480-3973-11eb-93d0-c99d5c896559.PNG">


After (create):
<img width="498" alt="transforms_error2" src="https://user-images.githubusercontent.com/7405507/101512976-89272b80-3973-11eb-9ef4-76ff6d267779.PNG">

Looking at the response to the `_preview` endpoint request in the network tab, you might have to add the full error to the attributes object (which @jgowdyelastic says is how it was done in the ML plugin).

Reply:
@peteharverson thanks for spotting this. With a new ES client, the default error message has been changed. I wrote an API integration test https://github.com/elastic/kibana/pull/84932/commits/9b06f7ecd4696ee292952d905d9da8fac4bc323e to validate this based on the initial behavior. And added the logic to extract error message from the root cause here https://github.com/elastic/kibana/pull/84932/commits/4099aa7d26946d54f0d51afafcdbbcb17311fa46